packages/apps/crm/src/screens/opportunity/OpportunityFormScreen.js

```diff
@@ -118,55 +127,90 @@ const OpportunityFormScreen = ({navigation, route}) => {
           <ClientProspectSearchBar
             titleKey="Crm_ClientProspect"
             placeholderKey="Crm_ClientProspect"
-            defaultValue={partner}
-            onChange={setPartner}
+            defaultValue={_opportunity.partner}
+            onChange={value => handleOpportunityFieldChange(value, 'partner')}
+            style={[styles.picker, styles.marginPicker]}
+            styleTxt={styles.marginTitle}
+            required={true}
+          />
+          <ContactSearchBar
+            titleKey="Crm_Contact"
+            placeholderKey="Crm_Contact"
+            defaultValue={_opportunity.contact}
+            onChange={value =>
+              handleOpportunityFieldChange({id: value?.id}, 'contact')
+            }
             style={[styles.picker, styles.marginPicker]}
             styleTxt={styles.marginTitle}
+            required={true}
           />
           <DateInput
             title={I18n.t('Crm_Opportunity_ExpectedCloseDate')}
-            defaultDate={date}
-            onDateChange={setDate}
+            defaultDate={
+              idOpportunity != null
+                ? new Date(_opportunity.expectedCloseDate)
+                : null
+            }
+            onDateChange={value =>
+              handleOpportunityFieldChange(
+                value?.toISOString().split('T')[0],
+                'expectedCloseDate',
+              )
+            }
             style={styles.input}
           />
           <FormIncrementInput
             title={I18n.t('Crm_Opportunity_Amount')}
-            defaultValue={amount}
-            onChange={setAmount}
+            defaultValue={idOpportunity != null ? _opportunity.amount : 0}
```

Comment:
To avoid all those checks when assigning default values of components, maybe you can initialized correctly empty opportunity with default value : 
```
const [_opportunity, setOpportunity] = useState(
    idOpportunity != null ? opportunity : {amount: 0, ...},
  );
```